### PR TITLE
New boolean condition for checking for multiple audio streams in the GUI

### PR DESCRIPTION
**Pull request by lobermann**
## Description
Added a new skin condition that allows checking if the currently playing item has more than 1 audio stream.

Can be used in the skin like this
```
<visible>VideoPlayer.HasMultipleAudioStreams</visible>
```

## Motivation and Context
I wanted to hide the audio stream selection button in the OSD when there is anyway just one audio stream available for the item that is currently played. 

## How Has This Been Tested?
I added the visible tag to the button in the OSD to hide it. Worked well on MacOS and LibreElec.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [X] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by ksooo — APPROVED** [review]
Nice. If you could come up with the respective Estuary changes, that would be great. Maybe @ronie can assist you?

**Comment by ronie** [comment]
sure, though i would prefer to disable the button instead of hiding it. 
`<enable>VideoPlayer.HasMultipleAudioStreams</enable>`

**Comment by lobermann** [comment]
Sure, I can PR that change to the skin. No problem. Will use the `<enable>` tag then.

**Comment by ksooo** [comment]
> Sure, I can PR that change to the skin. No problem. Will use the <enable> tag then.

 Cool. Thanks guys.

**Comment by ksooo** [comment]
Could we introduce a similar bool for multiple video streams?

**Comment by lobermann** [comment]
Ok, I just checked for the skin and there is another patch that I need to PR first.
Right now in Estuary there is no way of directly selecting audio or subtitles as  you have to go through the video settings from the OSD, which takes a few clicks.
I added two classes, `GUIDialogSubtitleSelect` and `GUIDialogAudioSelect` to have two buttons on the OSD so you can quickly access that and which just show the selection (similar as Netflix and other do) and no other settings. I will PR those dialogs in the coming days in a separate PR.

@ksooo yes, should be possible for video streams also, but have to check. But tbh I never saw a video that has multiple video streams. But I can check that out after the PR for the two dialogs.

**Comment by ksooo** [comment]
> But tbh I never saw a video that has multiple video streams.

 You should start using streaming services. 🤔 Multiple video Streams are very common with for example Youtube.

**Comment by ksooo** [comment]
> Right now in Estuary there is no way of directly selecting audio or subtitles as you have to go through the video settings from the OSD, which takes a few clicks.
I added two classes, GUIDialogSubtitleSelect and GUIDialogAudioSelect to have two buttons on the OSD so you 

 Not sure we actually have enough space left for another two buttons in Estuary's Player OSD. @ronie what do you think?

**Comment by jjd-uk** [comment]
> @ksooo yes, should be possible for video streams also, but have to check. But tbh I never saw a video that has multiple video streams. But I can check that out after the PR for the two dialogs.

If you need a local test file I've used this one from mplayer samples for testing the video stream switching feature, it's a short anime sample with 2 tracks of different quality levels and is 12M is size http://samples.mplayerhq.hu/Matroska/multiple_tracks.mkv

**Comment by ronie** [comment]
> Not sure we actually have enough space left for another two buttons in Estuary's Player OSD. @ronie what do you think?

indeed, no new buttons on the osd please. we tried to keep it clean by adding that settings button, which offers some other items.

**Comment by phunkyfish** [comment]
@lobermann do you want to progress this, close it or put it on the backburner?

**Comment by lobermann** [comment]
I do want to progress on this PR but it will take some time before I can get back to that.

**Comment by phunkyfish** [comment]
Cool, marking as PR cleanup: recent

**Comment by DaveTBlake** [comment]
This improvement is wanted @lobermann  but seems to have stalled. Bumping to v20  as v19 entering Beta phase with focus on testing and bug fixes.

**Comment by fuzzard** [comment]
@lobermann This has obviously fallen through the cracks. There looks to have been acceptance from both @ksooo and @ronie.

Multiple Video bool can be introduced in another PR. If you would like to give this one last shot at a rebase, and we'll get this in for v20

**Comment by lobermann** [comment]
@fuzzard thanks. Sure, will give it another rebase. Looking forward on getting this merged.

**Comment by enen92** [comment]
I am unsure if we actually need this. We have 3 types of infos: bools, ints and strings/labels.
The audiostream count is already available as an int info: `videoplayer.audiostreamcount`. This provides way more valuable information than a bool which simply checks if that number is bigger than 0.
Skins can easily use `Integer.*` complex conditions. I mean, the same that's being done here can already be done by a skin by using:

`Integer.IsGreater(VideoPlayer.AudioStreamCount,0)`

tldr: I find this addition redundant

**Comment by fuzzard** [comment]
@lobermann does the comment from enen satisfy the original intent for this PR?

**Comment by lobermann** [comment]
Yes, I agree with that. We can close this PR.

**Comment by fuzzard** [comment]
No worries. Thanks again for the PR, sorry it dragged on for so long.

**Comment by lobermann** [comment]
No problem at all! :-)
